### PR TITLE
Inject spring resources to wrapped processors

### PR DESCRIPTION
**Pull request by gurbuzali**
fixes https://github.com/hazelcast/hazelcast-jet/issues/988

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ProcessorWrapper.java [hunk 1]
```diff
@@ -28,12 +28,17 @@
  * processor.
  */
 public class ProcessorWrapper implements Processor {
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ProcessorWrapper.java]: should also be abstract class

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java [hunk 2]
```diff
@@ -190,8 +191,12 @@ private OutboxImpl createOutbox(@Nonnull OutboundCollector ssCollector) {
     @Override
     public void init() {
         if (serializationService.getManagedContext() != null) {
-            Object processor2 = serializationService.getManagedContext().initialize(processor);
-            assert processor2 == processor : "different object returned";
+            Processor toInit = processor;
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java]: can be replaced with `? :` expression

File: hazelcast-jet-spring/src/test/java/com/hazelcast/jet/spring/TestCustomProcessor.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    @BeforeClass
+    @AfterClass
+    public static void start() {
+        Jet.shutdownAll();
+    }
+
+    @Test
+    public void test() {
+        Pipeline p = Pipeline.create();
+        p.drawFrom(Sources.batchFromProcessor("source", preferLocalParallelismOne(CustomSourceP::new)))
```
Comment (cangencer) [inline hazelcast-jet-spring/src/test/java/com/hazelcast/jet/spring/TestCustomProcessor.java]: is this test actually testing the wrapping?

Comment (gurbuzali) [reply hazelcast-jet-spring/src/test/java/com/hazelcast/jet/spring/TestCustomProcessor.java]: not the source but the sink is wrapped

Comment (cangencer) [reply hazelcast-jet-spring/src/test/java/com/hazelcast/jet/spring/TestCustomProcessor.java]: OK, I see the sink is always wrapped.

**Pull request merged**